### PR TITLE
web: Move from many npm scripts, to a single build file

### PR DESCRIPTION
**Pull request by Dinnerbone**
Much more maintainable, and adding new options should be super easy now (such as webgpu).

It's also much faster, for me at least! There seems to be quite some seconds between every `npm run` on my machine.

### CODE REVIEW

File: web/packages/core/package.json [hunk 1]
```diff
@@ -10,26 +10,7 @@
...
-        "//7": "# Dispatches to either building the real, or copying the fake (stand-in),",
-        "//8": "# 'with-extensions' module.",
-        "build:wasm-extensions": "node -e \"process.exit(process.env.ENABLE_WASM_EXTENSIONS == 'true' ? 0 : 1)\" && npm run build:wasm-extensions-real || npm run build:wasm-extensions-fake",
-        "build:wasm-extensions-real": "echo \"Building module with WebAssembly extensions\" && cross-env OUT_NAME=ruffle_web-wasm_extensions CARGO_PROFILE=web-wasm-extensions RUSTFLAGS=\"$RUSTFLAGS --cfg=web_sys_unstable_apis -Aunknown_lints -C target-feature=+bulk-memory,+simd128,+nontrapping-fptoint,+sign-ext,+reference-types\" WASM_BINDGEN_FLAGS=\"--reference-types\" WASM_OPT_FLAGS=\"--enable-reference-types\" npm run build:wasm",
-        "build:wasm-extensions-fake": "echo \"Copying the vanilla module as stand-in\" && shx cp dist/ruffle_web_bg.wasm dist/ruffle_web-wasm_extensions_bg.wasm && shx cp dist/ruffle_web_bg.wasm.d.ts dist/ruffle_web-wasm_extensions_bg.wasm.d.ts && shx cp dist/ruffle_web.js dist/ruffle_web-wasm_extensions.js && shx cp dist/ruffle_web.d.ts dist/ruffle_web-wasm_extensions.d.ts",
```
Comment (relrelb) [inline web/packages/core/package.json:26]: `shx` is no longer used, thus can be removed from `devDependencies`.

**Review by relrelb — APPROVED** [review]
Looks good (apart from 1 minor comment)

File: web/packages/core/package.json (hunk 1 above)
Comment (Dinnerbone) [reply web/packages/core/package.json:26]: Good catch! Removed

File: web/packages/core/tools/build_wasm.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    });
+    console.log(`Running wasm-bindgen on ${flavor}...`);
+    runWasmBindgen({
+        path: `../../../target/wasm32-unknown-unknown/${profile}/ruffle_web.wasm`,
+        outName: filename,
+        dir: "dist",
+        flags: wasmBindgenFlags,
+    });
+    if (optimise) {
+        console.log(`Running wasn-opt on ${flavor}...`);
```
Comment (torokati44) [inline web/packages/core/tools/build_wasm.js:88]: Sorry, late, but typo: wasn-opt

**Pull request merged**